pyi.py

```diff
@@ -246,6 +246,31 @@ def _check_platform_check(self, node: ast.Compare) -> None:
         else:
             self.error(node, Y007)
 
+    def visit_FunctionDef(self, node: ast.FunctionDef) -> None:
+        self.generic_visit(node)
+
+        for i, statement in enumerate(node.body):
+            if i == 0:
+                # normally, should just be "..."
+                if isinstance(statement, ast.Pass):
+                    # should disable this if https://github.com/python/typeshed/pull/1009 isn't
+                    # accepted
+                    self.error(statement, Y009)
+                    continue
+                elif isinstance(statement, ast.Expr) and isinstance(statement.value, ast.Ellipsis):
+                    continue
+            # allow "raise", a number of stubs have this
+            if isinstance(statement, ast.Raise):
+                continue
+            # allow assignments in constructor for now
+            # (though these should probably be changed)
+            if node.name == '__init__' and isinstance(statement, ast.Assign) and \
+                    isinstance(statement.targets[0], ast.Attribute) and \
+                    isinstance(statement.targets[0].value, ast.Name) and \
+                    statement.targets[0].value.id == 'self':
+                continue
```

Comment:
A few things.

1. I dislike formatting that depends on backslashes. It would be cleaner as:
```py3
if (
	node.name == '__init__' and
	isinstance(statement, ast.Assign) and
	isinstance(statement.targets[0], ast.Attribute) and
	isinstance(statement.targets[0].value, ast.Name) and
	statement.targets[0].value.id == 'self'
):
	continue
```

2. More importantly, `self.x` can also be annotated with PEP-526 `AnnAssign`. This can be a separate if statement to keep the logic clean.
3. In the traditional `Assign` case, we want to prohibit assignments that *don't* use a type comment. So check for that, too.
4. You cannot assume `self`, for metaclasses we are using `cls` as the first argument in `__init__`. You should reuse the first argument name, whatever it is. In fact, in typeshed we have a few of those:
```
stdlib/2/decimal.pyi:57:    def __init__(cls, value: Union[_Decimal, float, str, unicode,
stdlib/3/decimal.pyi:129:    def __init__(cls, value: Union[_Decimal, float, str,
stdlib/3/email/policy.pyi:21:        def __init__(**kw: Any) -> None: ...
third_party/2/google/protobuf/reflection.pyi:7:    def __init__(cls, name, bases, dictionary) -> None: ...
third_party/2/yaml/__init__.pyi:40:    def __init__(cls, name, bases, kwds) -> None: ...
third_party/2and3/sqlalchemy/sql/visitors.pyi:8:    def __init__(cls, clsname, bases, clsdict) -> None: ...
third_party/2and3/sqlalchemy/util/langhelpers.pyi:134:    def __init__(cls, clsname, bases, clsdict) -> None: ...
```